src/vms/pvm/etna-builder/builder.ts

```diff
@@ -0,0 +1,1258 @@
+import { baseTxUnsafePvm, type SpendOptions, UnsignedTx } from '../../common';
+import { defaultSpendOptions } from '../../common/defaultSpendOptions';
+import type { Dimensions } from '../../common/fees/dimensions';
+import { addDimensions, createDimensions } from '../../common/fees/dimensions';
+import type { Context } from '../../context';
+import {
+  ID_LEN,
+  INTRINSIC_ADD_PERMISSIONLESS_DELEGATOR_TX_COMPLEXITIES,
+  INTRINSIC_ADD_PERMISSIONLESS_VALIDATOR_TX_COMPLEXITIES,
+  INTRINSIC_ADD_SUBNET_VALIDATOR_TX_COMPLEXITIES,
+  INTRINSIC_BASE_TX_COMPLEXITIES,
+  INTRINSIC_CREATE_CHAIN_TX_COMPLEXITIES,
+  INTRINSIC_CREATE_SUBNET_TX_COMPLEXITIES,
+  INTRINSIC_EXPORT_TX_COMPLEXITIES,
+  INTRINSIC_IMPORT_TX_COMPLEXITIES,
+  INTRINSIC_REMOVE_SUBNET_VALIDATOR_TX_COMPLEXITIES,
+  INTRINSIC_TRANSFER_SUBNET_OWNERSHIP_TX_COMPLEXITIES,
+  getAuthComplexity,
+  getInputComplexity,
+  getOutputComplexity,
+  getOwnerComplexity,
+  getSignerComplexity,
+} from '../txs/fee';
+import { spend } from './spend';
+
+const getAddressMaps = ({
+  inputs,
+  inputUTXOs,
+  minIssuanceTime,
+  fromAddressesBytes,
+}: {
+  inputs: readonly TransferableInput[];
+  inputUTXOs: readonly Utxo[];
+  minIssuanceTime: bigint;
+  fromAddressesBytes: readonly Uint8Array[];
+}): AddressMaps => {
+  return AddressMaps.fromTransferableInputs(
+    inputs,
+    inputUTXOs,
+    minIssuanceTime,
+    fromAddressesBytes,
+  );
+};
+
+const getMemoComplexity = (
+  spendOptions: Required<SpendOptions>,
+): Dimensions => {
+  return createDimensions(spendOptions.memo.length, 0, 0, 0);
+};
+
+/**
+ * Common properties used in all PVM transaction builder functions.
+ */
+type CommonTxProps = Readonly<{
+  /**
+   * List of addresses that are used for selecting which UTXOs are signable.
+   */
+  fromAddressesBytes: readonly Uint8Array[];
+  options?: SpendOptions;
+  /**
+   * List of UTXOs that are available to be spent.
+   */
+  utxos: readonly Utxo[];
+}>;
+
+type TxProps<T extends Record<string, unknown>> = CommonTxProps & Readonly<T>;
+
+type TxBuilderFn<T extends TxProps<Record<string, unknown>>> = (
+  props: T,
+  context: Context,
+) => UnsignedTx;
+
+export type NewBaseTxProps = TxProps<{
+  /**
+   * The desired output (change outputs will be added to them automatically).
+   */
+  outputs: readonly TransferableOutput[];
+}>;
+
+/**
+ * Creates a new unsigned PVM base transaction (`BaseTx`) using calculated dynamic fees.
+ *
+ * @param props {NewBaseTxProps}
+ * @param context {Context}
+ * @returns {UnsignedTx} An UnsignedTx.
+ */
+export const newBaseTx: TxBuilderFn<NewBaseTxProps> = (
+  { fromAddressesBytes, options, outputs, utxos },
+  context,
+) => {
+  const fromAddresses = addressesFromBytes(fromAddressesBytes);
+  const defaultedOptions = defaultSpendOptions(
+    [...fromAddressesBytes],
+    options,
+  );
+  const toBurn = new Map<string, bigint>([
+    [context.avaxAssetID, context.baseTxFee],
```

Comment:
No we are not.

Since we don't plan on using the builder for any pre-etna building of transactions, its safe to remove and is a carryover from copying AvalancheGo's builder logic that flips between static and dynamic calculators/calculations.

Reply:
Removed and simplified.